### PR TITLE
107 migrate existing completion provider

### LINKED ISSUE #7: As a developer, I want to autocomplete deps and srcs in my BUILD files
Provide autocomplete capabilities on dependency paths.

Notes:
- Just assume always in the deps rules
- Just do the most common case, `deps` parameter
- MORE ADVANCED: Use [this list](https://docs.bazel.build/versions/2.2.0/skylark/lib/attr.html) to actually determine what to autocomplete.

- [x] Use the analyzer to locate autocompletions.
- [x] Provide completion for both paths and rules.
- [x] Provide a range of possible completion items.
- [x] Completions should be sorted by most likely. E.g. given //path/to/mug, the resulting completion array will be something like

```
//path/to/mug <-- most likely
//path/to/muggles
//path/to/smuggles <-- least likely
etc
```

- [ ] Write test cases.

### LINKED ISSUE #107: Migrate existing completion provider to use new Workspace Tree

### PR DESCRIPTION
**Pull request by NovaMachina**
This migrates the completion provider over to the new workspace tree.

Note: For whatever reason I could not get automated tests to be consistent. I wold get a test to pass, move on to another test, and then the previous one would fail. But then I wold run the test a few more times and suddenly it would pass. I've tested it manually on VSCode and it works just fine.

### CODE REVIEW

File: server/src/main/java/server/completion/CompletionProvider.java [hunk 1]
```diff
@@ -1,158 +1,111 @@
...
 
-        } catch (IOException e) {
-            logger.error("Hit exception");
-            logger.error(Logging.stackTraceToString(e));
         } catch (Exception e) {
-            // TODO: Find the cause of the completion null pointer exception and error check accordingly
-            logger.error("Hit the catch for generic exceptions");
             logger.error(Logging.stackTraceToString(e));
-        }
+        }// TODO: Find the cause of the completion null pointer exception and error check accordingly
```
Comment (loganmcbride148) [inline server/src/main/java/server/completion/CompletionProvider.java:64]: Did you mean to leave this TODO? I'm not sure if it is something that must be fixed during this sprint.

**Review by loganmcbride148 — APPROVED** [review]
I left a comment in the CompletionProvider.java about a TODO. Other than that, everything looks and behavs well.

File: server/src/test/java/server/completion/CompletionProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+//                        "        \"//third_party/java:lsp4j\",\n" +
+//                        "        \"//third_party/java:mockito\",\n" +
+//                        "        \"//third_party/java:powermock\",\n" +
+//                        "        \"//third_party/java:powermock-junit\",\n" +
+//                        "        \"//third_party/java:powermock-mockito\",\n" +
+//                        "        \"/\"\n" +
+//                        "    ],\n" +
+//                        ")\n").when(tracker).getContents(Mockito.any());
+//        CompletionParams params = new CompletionParams(new TextDocumentIdentifier("somedocument"), new Position(26,10), new CompletionContext(CompletionTriggerKind.TriggerCharacter, "/"));
+//        CompletableFuture<Either<List<CompletionItem>, CompletionList>> future = classUnderTest.getCompletion(params);
```
Comment (josiahsrc) [inline server/src/test/java/server/completion/CompletionProviderTest.java]: I feel like the test issues have something to do with static instances

**Review by josiahsrc — APPROVED** [review]
Would be good to fix up those tests, but other than that looks good

File: server/src/test/java/server/completion/CompletionProviderTest.java (hunk 2 above)
Comment (NovaMachina) [reply server/src/test/java/server/completion/CompletionProviderTest.java]: I've now removed all the static instances but still having a similar issue. Its weird

**Pull request merged**